internal/js/modules/k6/grpc/client.go

```diff
@@ -387,6 +397,71 @@ func (c *Client) buildInvokeRequest(
 	}, nil
 }
 
+// normalizeNumberStrings recursively traverses a sobek.Value. It creates a deep copy of any
+// objects or arrays, and converts special float values (NaN, Infinity) to their
+// string representations for proper JSON serialization.
+func normalizeNumberStrings(v sobek.Value, runtime *sobek.Runtime, stack []*sobek.Object) (sobek.Value, error) {
+	// If v is a primitive (not an object or array), handle it directly.
+	if v == nil || sobek.IsNull(v) || sobek.IsUndefined(v) {
+		return v, nil
+	}
+
+	// Handle special float values.
+	if f, ok := v.Export().(float64); ok {
```

Comment:
As @mstoykov suggested before, could we use `ExportType()` here instead of `Export()`, as the latter is a much more expensive operation.

In case it's a `float64`, I guess you'll need to call `v.Export()` anyway, but in all the other cases, it won't be called that early.

Reply:
Or maybe even something like:

```suggestion
	if v.ExportType().Kind() == reflect.Float64 {
		f := v.ToFloat()
		switch {
		case math.IsNaN(f):
			return runtime.ToValue("NaN"), nil
		case math.IsInf(f, 1):
			return runtime.ToValue("Infinity"), nil
		case math.IsInf(f, -1):
			return runtime.ToValue("-Infinity"), nil
		}

		return v, nil
	}
```

So, you don't need `Export()` at all.
Wdyt? @mstoykov

Reply:
This seems better.

For the record - the idea is that if you are using `Export` it will do a lot of work in some of the cases. If the end type is flaot64 - that probably is fine. But it also might be a big object that you just fully transformed from JS to go type. And then you just throw it away because it isn't a float64, just to repeat the operation each time.

Reply:
Sure, no problem @ariasmn, that's why we have a review process, we're all humans and sometimes it happens!

Looks good now! 👌🏻 Thanks! 🙌🏻